pkg/controller/common/util_test.go

```diff
@@ -300,6 +300,52 @@ var _ = Describe("Rebind", func() {
 	})
 })
 
+var _ = Describe("GetMetricsURL", func() {
+	makePod := func(ip string, withMetrics bool) *v1.Pod {
+		pod := &v1.Pod{
+			Status: v1.PodStatus{
+				PodIP: ip,
+			},
+		}
+
+		if !withMetrics {
+			return pod
+		}
+
+		pod.Spec = v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{
+						{Name: "metrics", ContainerPort: 8080},
+					},
+				},
+			},
+		}
+
+		return pod
+	}
+
+	It("Should suceed with IPv4", func() {
+		pod := makePod("127.0.0.1", true)
+		url, err := GetMetricsURL(pod)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(url).To(Equal("https://127.0.0.1:8080/metrics"))
+	})
+
+	It("Should suceed with IPv6", func() {
+		pod := makePod("::1", true)
+		url, err := GetMetricsURL(pod)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(url).To(Equal("https://[::1]:8080/metrics"))
+	})
+
+	It("Should fail when there is no metrics port", func() {
+		pod := makePod("127.0.0.1", false)
+		_, err := GetMetricsURL(pod)
+		Expect(err).To(HaveOccurred())
```

Comment:
Just a nit, if you want to, and since you know exactly which error to expect, you could assert
that you indeed get back this specific error
`Expect(err).To(MatchError(os.ErrNotExist))`

Reply:
It is actually a regular string error: https://github.com/kubevirt/containerized-data-importer/blob/afb269737bfc90cad4f7eccfdf6d157a60b2c482/pkg/controller/common/util.go#L1632

Reply:
Yeah, you'd have to take it out to a var to perform the assertion

Reply:
Not sure it's worth the hassle then